diff --git a/PKGBUILD b/PKGBUILD
index 5c18fe7..470b299 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -13,11 +13,9 @@ _url=https://git.kernel.org/pub/scm/libs/libcap/libcap.git
 license=('BSD-3-Clause OR GPL-2.0-only')
 depends=(
   glibc
-  pam
 )
 makedepends=(
   git
-  go
   linux-api-headers
 )
 provides=(
@@ -66,6 +64,8 @@ build() {
     lib=lib
     prefix=/usr
     sbindir=bin
+    PAM_CAP=no
+    GOLANG=no
     -C $pkgname
   )
 
@@ -96,6 +96,8 @@ package() {
     lib=lib
     prefix=/usr
     sbindir=bin
+    PAM_CAP=no
+    GOLANG=no
     install
     -C $pkgname
   )
